src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -557,8 +558,22 @@ bool dcd_edpt_open (uint8_t rhport, tusb_desc_endpoint_t const * desc_edpt)
   xfer->max_size = desc_edpt->wMaxPacketSize.size;
   xfer->interval = desc_edpt->bInterval;
 
+  uint16_t const fifo_size = tu_max16((desc_edpt->wMaxPacketSize.size + 3) / 4, 16); // Round up to next full word, minimum value must be 16
+
   if(dir == TUSB_DIR_OUT)
   {
+    // Calculate required size of RX FIFO
+    uint16_t const sz = 15 + 2*fifo_size + 2*EP_MAX;
```

Comment:
Good idea, i called it
`static inline uint16_t calc_rx_ff_size(uint16_t ep_size)
{
  return 15 + 2*ep_size + 2*EP_MAX;
}
`
The minimum 16 is not required since this is handled outside.